atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweepTaskRunnerTest.java

```diff
@@ -188,57 +188,124 @@ public void testDontSweepFutureConservative() {
         assertEquals(2, results.getStaleValuesDeleted());
         // Future timestamps don't count towards the examined count
         assertThat(results.getCellTsPairsExamined()).isGreaterThanOrEqualTo(3);
-        assertEquals(ImmutableSet.of(-1L, 100L, 125L, 150L), getAllTs("foo"));
+        assertEquals(ImmutableSet.of(-1L, 100L, 125L, 150L), getAllTsFromDefaultColumn("foo"));
     }
 
-    @Test
+    @Test(timeout = 50000)
     public void testSweepOneThorough() {
         createTable(SweepStrategy.THOROUGH);
         putIntoDefaultColumn("foo", "bar", 50);
         putIntoDefaultColumn("foo", "baz", 100);
         SweepResults results = completeSweep(175);
         assertEquals(1, results.getStaleValuesDeleted());
         assertThat(results.getCellTsPairsExamined()).isGreaterThanOrEqualTo(2);
-        assertEquals("baz", get("foo", 150));
-        assertNull(get("foo", 80));
-        assertEquals(ImmutableSet.of(100L), getAllTs("foo"));
+        assertEquals("baz", getFromDefaultColumn("foo", 150));
+        assertNull(getFromDefaultColumn("foo", 80));
+        assertEquals(ImmutableSet.of(100L), getAllTsFromDefaultColumn("foo"));
     }
 
-    @Test
+    @Test(timeout = 50000)
     public void testDontSweepLatestThorough() {
         createTable(SweepStrategy.THOROUGH);
         putIntoDefaultColumn("foo", "bar", 50);
         SweepResults results = completeSweep(75);
         assertEquals(0, results.getStaleValuesDeleted());
         assertThat(results.getCellTsPairsExamined()).isGreaterThanOrEqualTo(1);
-        assertEquals("bar", get("foo", 150));
-        assertEquals(ImmutableSet.of(50L), getAllTs("foo"));
+        assertEquals("bar", getFromDefaultColumn("foo", 150));
+        assertEquals(ImmutableSet.of(50L), getAllTsFromDefaultColumn("foo"));
     }
 
-    @Test
+    @Test(timeout = 50000)
+    public void testSweepLatestDeletedMultiRowThorough() {
+        createTable(SweepStrategy.THOROUGH);
+        putIntoDefaultColumn("foo", "", 50);
+        put("foo-2", "other", "womp", 60);
+        SweepResults results = completeSweep(75);
+        assertEquals(1, results.getStaleValuesDeleted());
+        assertThat(results.getCellTsPairsExamined()).isGreaterThanOrEqualTo(1);
+
+        assertNull(getFromDefaultColumn("foo", 150));
+        assertEquals(ImmutableSet.of(), getAllTsFromDefaultColumn("foo"));
```

Comment:
we may want to run another iteration of sweep here, to verify it correctly does nothing if all values are already deleted

Reply:
Added on another of the new tests, where we actually have nothing left in the table. There are also a couple of tests (for conservative sweep) where we sweep tables twice.